packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -411,3 +415,40 @@ getCollectionHooks("Posts").updateAfter.add(async (post: DbPost, props: CreateCa
 
   return post;
 });
+
+getCollectionHooks("Posts").createAfter.add(async (post: DbPost, props: CreateCallbackProperties<DbPost>) => {
+  await addPostToSequence(post, props);
+  return post
+});
+
+getCollectionHooks("Posts").updateAfter.add(async (post: DbPost, props: CreateCallbackProperties<DbPost>) => {
+  await addPostToSequence(post, props);
+  return post
+})
+
+async function addPostToSequence(post: DbPost, props: CreateCallbackProperties<DbPost>) {
+  // If the post has a canonicalSequenceId and isn't in the sequence, add it to the sequence's last chapter.
+  // This is useful for adding draft posts to sequences by adding the canonicalSequenceId on the Post edit page,
+  // because you can't find draft posts with the Add/Remove Posts functionality on a Sequence page.
+  if (!post.canonicalSequenceId) return post;
+
+  const {currentUser, context} = props;
+  if (await sequenceContainsPost(post.canonicalSequenceId, post._id, context)) return post;
+
+  const sequence = await Sequences.findOne({ _id: post.canonicalSequenceId });
+  if (!Sequences.options.mutations.edit.check(currentUser, sequence)) return null;
+
+  const chapters = await mongoFind("Chapters", {sequenceId: post.canonicalSequenceId}, {sort: {number: 1}});
```

Comment:
Can this just be `Chapters.find(...}.fetch()`?

Reply:
We have very few uses of `mongoFind` in the codebase and I'd probably rather phase it out entirely